src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1464,14 +1464,19 @@ tx_gateway_frontend::do_commit_tm_tx(
         }
 
         std::vector<ss::future<prepare_tx_reply>> pfs;
-        for (auto rm : tx.partitions) {
-            pfs.push_back(_rm_partition_frontend.local().prepare_tx(
-              rm.ntp,
-              rm.etag,
-              model::tx_manager_ntp.tp.partition,
-              tx.pid,
-              tx.tx_seq,
-              timeout));
+        // We can ignore prepare phase for rm_stm. Becasue we decided use
+        // quorum_ack for all data records on rm_stm. And we do not need marker
+        // to sync replicas
+        if (!is_transaction_ga()) {
```

Comment:
another thing to get rid from txn coordinator is to remove just_started branch in add_partition and add_group